File: 백준/Silver/2606. 바이러스/바이러스.js
const fs = require("fs");
const input = fs.readFileSync("/dev/stdin", "utf8").trim().split("\n");
const n = parseInt(input[0], 10);
const m = parseInt(input[1], 10);
const edges = input.slice(2).map(line => line.split(" ").map(Number));
const graph = Array.from({ length: n + 1 }, () => []); 
const visited = Array(n + 1).fill(false);
edges.forEach(([a, b]) => {
  graph[a].push(b);
  graph[b].push(a); 
});

let infectionCnt = 0;

const dfs = (node) => {
  visited[node] = true;
  for (let next of graph[node]) {
    if (!visited[next]) {
      infectionCnt++;
      dfs(next)
    }
  }
}

dfs(1)
console.log(infectionCnt)

File: 백준/Silver/2606. 바이러스/바이러스.py
from collections import deque
import sys

input = sys.stdin.readline

N = int(input())
Link = int(input())
graph = [[] for _ in range(N+1)]
for _ in range(Link):
    a, b = map(int, input().split())
    graph[a].append(b)
    graph[b].append(a)
visited = [False] * (N+1)
def bfs():
    queue = deque([1])
    visited[1] = True
    count = 0
    while queue:
        current = queue.popleft()
        for next in graph[current]:
            if not visited[next]:
                visited[next] = True
                queue.append(next)
                count+=1
    print(count)
bfs()
